cmd/snap-update-ns/utils.go

```diff
@@ -78,7 +117,41 @@ type Secure struct{}
 // things in $SNAP_DATA or in /tmp but we don't want to do so in /etc or in
 // other sensitive places.
 func (sec *Secure) CheckTrespassing(fd int, segments []string, segNum int) error {
-	return nil
+	// If the full path is allowed there's no problem. We don't go and check
+	// the filesystem type because it is just going to spam tests with extra
+	// fstatfs calls and it doesn't change anything.
+	p := "/" + strings.Join(segments, "/") + "/"
```

Comment:
Or `(path|filepath).Join()` unless extra `.Clean()` is a bother.

BTW. I think it's enough to `strings.Join(segments[:segNum+1])`